src/library/Navigation/Navigation.js

```diff
@@ -0,0 +1,173 @@
+      moreLabel: 'More navigation items',
+      moreText: 'More'
+    }
+  };
+
+  render() {
+    const {
+      align,
+      items: ignoreItems,
+      minimal,
+      secondary,
+      type: typeProp,
+      ...restProps
+    } = this.props;
+
+    const prefix = secondary ? PREFIX.secondary : PREFIX.primary;
+    const type = typeProp
+      ? `_${typeProp}`
+      : minimal
+        ? INTERNAL_TYPE.minimal
+        : INTERNAL_TYPE.none;
+
+    const rootProps = {
+      align,
+      prefix,
+      type,
+      ...restProps
+    };
+
+    return (
+      <NavigationRoot {...rootProps}>
+        {this.renderFromData({ prefix, type })}
+      </NavigationRoot>
+    );
+  }
+
+  renderFromData = (typeAndPrefix: Object) => {
+    const {
+      children,
+      itemElement,
+      items: itemsProp,
+      messages,
+      overflowAtIndex
+    } = this.props;
+
+    const items = children
+      ? Children.map(children, (child) => {
+          const { children, ...rest } = child.props;
+          return {
+            child,
+            text: children,
+            ...rest
+          };
+        })
+      : itemsProp;
+
+    let overflowData;
+    if (items && overflowAtIndex) {
+      overflowData = items.slice(overflowAtIndex);
+    }
+
+    return (
+      items &&
+      items.length &&
+      items.map(({ child, text, ...restItem }, index) => {
+        if (overflowAtIndex && index >= overflowAtIndex) {
+          if (index === overflowAtIndex) {
+            const menuProps = {
+              data: overflowData,
+              onClick: this.handleClick,
+              index,
+              itemElement,
+              messages,
+              ...typeAndPrefix
+            };
+
+            return <NavOverflowMenu key={index} {...menuProps} />;
+          } else {
+            return;
+          }
+        } else {
+          const navItemProps = {
+            ...this.getNavItemProps({
+              index,
+              ...restItem,
+              ...typeAndPrefix
+            }),
+            children: text
+          };
+
+          return child ? (
+            cloneElement(child, { key: index, ...navItemProps })
+          ) : (
+            <NavItem key={index} {...navItemProps} />
+          );
+        }
+      })
+    );
+  };
+
+  getNavItemProps: GetNavItemProps = ({
+    disabled,
+    element,
+    index,
+    onClick,
+    maxWidth,
+    selected,
+    ...restItemProps
+  }) => {
+    const {
+      itemElement,
+      items: ignoreItems,
+      maxItemWidth,
+      onChange: ignoreOnChange,
+      selectedIndex,
+      type: ignoreType,
+      ...restNavProps
+    } = this.props;
+
+    return {
+      disabled,
+      element: element || itemElement,
+      index,
+      maxWidth: maxWidth || maxItemWidth,
+      onClick: !disabled
+        ? composeEventHandlers(onClick, this.handleClick)
+        : // If we render `a` with an `href` for items, we need to preventDefault() to "disable" it
+          (event) => {
```

Comment:
Could extract a `preventDefaultEventFn` or similar so a new function is not created each time.